src/Traversals/depthfirst.jl

```diff
@@ -142,3 +142,123 @@ end
 end
 
 @traitfn is_cyclic(g::::(!IsDirected), alg::DepthFirst=DepthFirst()) = ne(g) > 0
+
+
+
+mutable struct Biconnections{E <: AbstractEdge} <: AbstractTraversalState
```

Comment:
`AbstractTraversalState` will been renamed to `TraversalState` when #1335 is merged.

This should be renamed `BiconnectionState`.